hyperopt/spark.py

```diff
@@ -126,20 +126,19 @@ def _decide_parallelism(max_num_concurrent_tasks, parallelism):
         elif parallelism <= 0:
             logger.warning(
                 "User-specified parallelism was invalid value ({p}), so parallelism "
-                "will be set to max_num_concurrent_tasks ({c}).".format(
+                "will be set to max number of concurrent tasks ({c}).".format(
                     p=parallelism, c=max_num_concurrent_tasks
                 )
             )
             parallelism = max_num_concurrent_tasks
         elif parallelism > max_num_concurrent_tasks:
             logger.warning(
-                "User-specified parallelism ({p}) is greater than "
-                "max_num_concurrent_tasks ({c}), so parallelism will be set to "
-                "max_num_concurrent_tasks.".format(
+                "User-specified parallelism ({p}) is greater than the max number of concurrent "
+                "tasks ({c}) this cluster can run now. If dynamic allocation is enabled for the "
+                "cluster, you might see more executors allocated.".format(
                     p=parallelism, c=max_num_concurrent_tasks
                 )
             )
-            parallelism = max_num_concurrent_tasks
```

Comment:
@maxpumperla Not only logging changed. See this line, I remove the parallelism limit here. So that, when spark dynamic allocation enabled, when user-specified parallelism larger than current spark max number of concurrent tasks, it will allow spark to allocate more executors (The previous code will limit parallelism to be max_num_concurrent_tasks, which prevent spark to allocate more executors)